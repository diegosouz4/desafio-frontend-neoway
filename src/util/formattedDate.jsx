export default function formattedDate(date) {
  if (!date) return;

  const newDate = date.split("T")[0].replaceAll("-", "/");
  return newDate;
}
