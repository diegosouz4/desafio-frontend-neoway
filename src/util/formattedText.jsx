export default function formattedText(text, limit = 50) {
  if (!text || text.length < limit) return text;
  let formattedText = text.substring(0, limit);
  formattedText = `${formattedText}...`;
  return formattedText;
}
