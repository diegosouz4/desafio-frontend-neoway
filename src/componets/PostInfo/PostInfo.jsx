import PropTypes from "prop-types";
import formattedDate from "../../util/formattedDate";
import { FaBookmark, FaCalendar, FaUser } from "react-icons/fa6";
import formattedText from "../../util/formattedText";

export default function PostInfo({ publishedAt, author, className, likes }) {
  const newDate = formattedDate(publishedAt) || null;
  const formattedAuthor = formattedText(author, 15) || null;

  console.log(likes);

  return (
    <div className={className}>
      {newDate && (
        <span>
          <FaCalendar /> {newDate}
        </span>
      )}
      {formattedAuthor && (
        <span>
          <FaUser /> {formattedAuthor}
        </span>
      )}
      {likes && (
        <span>
          <FaBookmark /> Salvo
        </span>
      )}
    </div>
  );
}

PostInfo.propTypes = {
  author: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
  className: PropTypes.string,
  likes: PropTypes.bool,
};
