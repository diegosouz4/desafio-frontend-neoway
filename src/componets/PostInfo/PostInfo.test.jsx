import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import PostInfo from "./PostInfo";
import formattedDate from "../../util/formattedDate";
import formattedText from "../../util/formattedText";
import "@testing-library/jest-dom";

const mockInfo = {
  author: "Jest",
  publishedAt: "2023-09-30T13:55:36Z",
  className: "jest-custom-class",
  likes: true,
};

describe("<PostInfo />", () => {
  it("should be render", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <PostInfo {...mockInfo} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should title be formatted", () => {
    const text = "This is a long text that needs to be formatted";
    const formattedTitle = formattedText(text, 15);

    render(
      <BrowserRouter>
        <PostInfo
          publishedAt={mockInfo.publishedAt}
          author={formattedTitle}
          likes={mockInfo.likes}
        />
      </BrowserRouter>
    );

    const author = screen.getByText("This is a long ...");
    expect(author).toBeInTheDocument();
  });

  it("should data to be formatted", () => {
    const dateFormatted = formattedDate(mockInfo.publishedAt);

    render(
      <BrowserRouter>
        <PostInfo {...mockInfo} />
      </BrowserRouter>
    );

    const renderDate = screen.getByText(dateFormatted);
    expect(renderDate).toBeInTheDocument();
  });
});
