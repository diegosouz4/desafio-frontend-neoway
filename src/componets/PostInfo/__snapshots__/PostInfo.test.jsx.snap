// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostInfo /> should be render 1`] = `
<DocumentFragment>
  <div
    class="jest-custom-class"
  >
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
        />
      </svg>
       2023/09/30
    </span>
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
        />
      </svg>
       Jest
    </span>
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 48V487.7C0 501.1 10.9 512 24.3 512c5 0 9.9-1.5 14-4.4L192 400 345.7 507.6c4.1 2.9 9 4.4 14 4.4c13.4 0 24.3-10.9 24.3-24.3V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48z"
        />
      </svg>
       Salvo
    </span>
  </div>
</DocumentFragment>
`;
